webpack/assets/javascripts/react_app/components/common/forms/ForemanForm/ForemanForm.js

```diff
@@ -4,12 +4,19 @@ import PropTypes from 'prop-types';
 import Form from '../Form';
 import { translate as __ } from '../../../../common/I18n';
 
-export const prepareErrors = errors =>
-  Object.keys(errors).reduce((memo, key) => {
-    const errorMessages = errors[key];
-    memo[key] = errorMessages ? errorMessages.join(', ') : errorMessages;
-    return memo;
-  }, {});
+export const prepareErrors = exception => {
+  const { errors } = exception;
+  try {
+    return Object.keys(errors).reduce((memo, key) => {
+      const errorMessages = errors[key];
+      memo[key] = errorMessages ? errorMessages.join(', ') : errorMessages;
+      return memo;
+    }, {});
+  } catch (e) {
+    // so we don't swallow non-form-related JS errors
+    throw exception;
```

Comment:
now that the errors object changed, e.g:
```js
errors: {
        name: ['is already taken', 'is too short'],
        email: ['is not a valid format'],
        phone: ['is too long'],
      },
```
would it affect other form consumers?
I think it worth to add some documentation

Reply:
@laviro I don't think the shape of the errors object is changed (unless I'm missing something?)  The object is handled the same way as before.. the only difference is it catches previously swallowed errors during the `prepareErrors` function.  (see my [reply here](https://github.com/theforeman/foreman/pull/7265#discussion_r363443565))